### PR TITLE
Updates to Save and Reload Use Case

### PR DESCRIPTION
**Pull request by djesurumc**
- Updated ReloadGameInteractor to remove method parameter (not injecting Gateway into the method but into the constructor)
- Created test for SaveGameInteractor (tested whether saving game runs without errors)
- Created test for ReloadGameInteractor (tested whether the information from a previous GameState is consistent with the new GameState when reloaded)

### CODE REVIEW

**Review by ethanhonest — APPROVED** [review]
Added necessary documentation and testing. Important change

**Review by petesaiacct — APPROVED** [review]
Good improved documentation. The game interactor test is a crucial addition, nice to see it implimenteted.

File: src/test/java/save_and_restart_use_case/ReloadGameInteractorTest.java [hunk 1]
```diff
@@ -0,0 +1,70 @@
+package save_and_restart_use_case;
+
+import CoreEntities.IO.DictionaryDataReaderGateway;
+import CoreEntities.IO.Presenter;
+import core_entities.game_parts.BoardFactory;
+import core_entities.game_parts.DefaultBoardFactory;
+import core_entities.game_parts.GameState;
+import default_reference_values.DefaultBoardDataAccessObject;
+import launch_new_game_use_case.CreateDictionaryDataAccessObject;
```
Comment (JosephSun2003) [inline src/test/java/save_and_restart_use_case/ReloadGameInteractorTest.java:9]: alot of these imports can be just replaced with a "packages.*;" instead. not required.

**Review by JosephSun2003 — APPROVED** [review]
Minor optimizations can be made but otherwise should function as intended.

**Pull request merged**